src/context/VideoContext.jsx

```diff
@@ -0,0 +1,39 @@
+import {useReducer, createContext, useContext, useEffect, useState} from "react";
+import { videoReducer } from "../reducer/VideoReducer";
+import { ByCategoryName, FilterByCategory  } from "../Utils/index";
+import axios from "axios";
+
+const VideoContext = createContext();
+
+const VideoProvider = ({children}) => {
+
+    const initialState = {videos: [], categoryName: "ALL"}
+    const [videoState, videoDispatch] = useReducer(videoReducer, initialState)
+    const {videos, categoryName} = videoState;
+    const [videoSearch, getvideoSearch] = useState("")
+
+    useEffect(() => {
+        (async () => {
+            const {status, data: {videos},} = await axios.get("/api/videos")
+
+            if(status === 200) {
+                videoDispatch({type: "GET_VIDEOS", payload: videos})
+            }
+        })();
```

Comment:
You can do error handling in case the expected status is not received. Rest the code is well written and clean. You can merge